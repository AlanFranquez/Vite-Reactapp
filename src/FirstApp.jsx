import Proptypes from 'prop-types';
import React from 'react'

// const obj = {
//     nombre: 'Alan',
//     age: 20
// }

// Siempre que se pueda intentar poner las funciones y variables por fuera de la funci√≥n para mejorar el rendimiento

// function Hola(a,b) {
//     return a+b
// }

function FirstApp({titulo, subtitulo}) {


  return (
    <>
    {/* <h1>{Hola(1,4)}</h1> */}
    {/* <code>{JSON.stringify(obj)}</code> */}
   
    <h1>{titulo}</h1>
    <h2>{subtitulo +1}</h2>
    <p>Lorem ipsum dolor sit amet consectetur, adipisicing elit. Animi accusamus impedit quaerat itaque harum, soluta laborum eaque, et tenetur dolores nulla vel rem voluptatum minus rerum vitae quae, veritatis voluptatibus?</p>
    </>
  )
}

//Proptypes, obligamos a los desarolladores a colocarle valores a los diferentes props.
FirstApp.propTypes = {
  titulo: Proptypes.string.isRequired,
  subtitulo: Proptypes.number.isRequired
}

export default FirstApp