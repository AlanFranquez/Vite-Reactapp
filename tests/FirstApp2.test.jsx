import { render, screen } from "@testing-library/react"
import { title } from "process";
import FirstApp from "../src/FirstApp"

describe('Pruebas sobre FirstApp', () => {

    const titulo = 'Hola Mundo'
    const subtitulo = 'prueba'

    test('Debe de hacer match con el snapshot', () => {

        const {container} = render(<FirstApp titulo={titulo}/>)

        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje "Hola Mundo"', () => {

        
        render(<FirstApp titulo={titulo}/>)
        expect(screen.getByText(titulo)).toBeTruthy()
        //screen.debug()
    });

    test('Debe de mostrar el titulo en un h1', () => {

        render(<FirstApp titulo={titulo}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(titulo)

    });

    test('Debe de mostrar el subtitulo enviado por las props', () => {
        render(<FirstApp titulo={titulo} subtitulo={subtitulo}/>);

    expect(screen.getAllByText(subtitulo).length).toBe(2);
    });
    
})