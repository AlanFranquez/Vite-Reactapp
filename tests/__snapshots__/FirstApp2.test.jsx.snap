// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola Mundo
     
  </h1>
  <h2>
    prueaaaba
  </h2>
  <h2>
    prueaaaba
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Animi accusamus impedit quaerat itaque harum, soluta laborum eaque, et tenetur dolores nulla vel rem voluptatum minus rerum vitae quae, veritatis voluptatibus?
  </p>
  <p>
    Alan Fr√°nquez
  </p>
</div>
`;
